repeat_expansion_variants/clinvar_identifier_parsing.py

```diff
@@ -0,0 +1,108 @@
+# expansion variant might be addressed as c.87+14_c.87+17. In this case, we don't want the pivots (87), but only the
+# actual coordinates (+14 and +17). To do that, we have a regular expression which captures the pivot part.
+coordinate_pivot_part = (
+    r'(?:'       # Non-capturing group for coordinate pivot part
+    r'[-+]?'     # Coordinate pivot may start with a minus or plus
+    r'[0-9]+'    # Then it contains a number
+    r'(?=[-+])'  # It then must be followed with either a plus or minus.  This is a positive lookahead and is *not*
+                 # part of the coordinate pivot, hence the (?=...) notation
+    r')?'        # The coordinate pivot is always optional
+)
+
+# Pattern 1. HGVS-like notation for coding or genomic coordinates. This pattern is used for most variants. From it we
+# can always extract transcript ID and start coord, and sometimes also end coord and repeat unit sequence.
+# Example: 'NM_001256054.2(C9orf72):c.-45+63_-45+80GGGGCC(2_25)'
+re_hgvs_like_transcript_or_genomic = re.compile(
+    hgvs_like_transcript_part +      # Transcript definition                                NM_001256054.2(C9orf72):
+    r'[gc]'                          # Coordinate type, genomic or coding                   c
+    r'\.'                            # Delimiter, coordinate type/coordinate                .
+
+    + coordinate_pivot_part +        # Start coordinate pivot, optional                     -45
+    r'\*?'                           # Sometimes there is an asterisk in front of the
+                                     # coordinate (special case, not important)
+    r'(?P<start_coord>[+-]?[0-9]+)'  # Start coordinate                                     +63
+
+    r'(?:'                           # Non-capruting group for end coordinate
+    r'_'                                 # Delimiter: start coordinate/end coordinate       _
+    + coordinate_pivot_part +            # End coordinate pivot, optional                   -45
+    r'\*?'                               # Coordinate may start with an asterisk
+    r'(?P<end_coord>[+-]?[0-9]+)'        # End coordinate                                   +80
+    r')?'                            # The entire end coordinate part is optional
+
+    r'(?P<sequence>[ATGC]*)'         # Repeat unit sequence, optional                       GGGGCC
+)
+
+# Pattern 2. HGVS-like notation for protein coordinates. This pattern is used for a few variants. From it we do not
+# extract any useful information, but always assume that such a notation signified a trinucleotide repeat type.
+# Example: NP_002964.3:p.Gln166(>=33)
+re_hgvs_like_protein = re.compile(hgvs_like_transcript_part + r'p\.')
+
+# Pattern 3. Human-readable description notation. This pattern is used for about a dozen variants. From it we only
+# extract the repeat unit sequence. Example: 'ATXN8, (CAG)n REPEAT EXPANSION'
+re_description = re.compile(
+    r'\('
+    r'(?P<sequence>[ATGC]+)'
+    r'\)n'
+    r' REPEAT EXPANSION'
+)
```

Comment:
Have you validated these assumptions for human readable representations across Clinvar datasets over time?

Reply:
Actually yes, they do change from time to time. The next PR, #4, includes a small fix for this specific part (they renamed "REPEAT EXPANSION" to just "EXPANSION"). However, I'm afraid there is nothing we can do here, as we cannot foresee how ClinVar will update the names in the future.

To help catch such changes, there is functionality (I can't remember if it's implemented in this PR or #4) to output all names which could not be parsed to a log file, and there is a point in review checklist to manually verify that there are no new unparsed names compared to the previous batch.